types/v56/types.go

```diff
@@ -2240,31 +2242,40 @@ type QueryResultEdgeGatewayRecordType struct {
 // QueryResultVMRecordType represents a VM record as query result.
 type QueryResultVMRecordType struct {
 	// Attributes
-	HREF                    string `xml:"href,attr,omitempty"`       // The URI of the entity.
-	Name                    string `xml:"name,attr,omitempty"`       // VM name.
-	Deployed                bool   `xml:"isDeployed,attr,omitempty"` // True if the virtual machine is deployed.
-	Status                  string `xml:"status,attr,omitempty"`
-	Busy                    bool   `xml:"isBusy,attr,omitempty"`
-	Deleted                 bool   `xml:"isDeleted,attr,omitempty"`
-	MaintenanceMode         bool   `xml:"isInMaintenanceMode,attr,omitempty"`
-	Published               bool   `xml:"isPublished,attr,omitempty"`
-	VAppTemplate            bool   `xml:"isVAppTemplate,attr,omitempty"`
-	VdcEnabled              bool   `xml:"isVdcEnabled,attr,omitempty"`
-	VdcHREF                 string `xml:"vdc,attr,omitempty"`
-	VAppParentHREF          string `xml:"container,attr,omitempty"`
-	VAppParentName          string `xml:"containerName,attr,omitempty"`
-	HardwareVersion         int    `xml:"hardwareVersion,attr,omitempty"`
-	HighestSupportedVersion int    `xml:"pvdcHighestSupportedHardwareVersion,attr,omitempty"`
-	VmToolsVersion          string `xml:"vmToolsVersion,attr,omitempty"`
-	GuestOS                 string `xml:"guestOs,attr,omitempty"`
-	MemoryMB                int    `xml:"memoryMB,attr,omitempty"`
-	Cpus                    int    `xml:"numberOfCpus,attr,omitempty"`
-	StorageProfileName      string `xml:"storageProfileName,attr,omitempty"`
-	NetworkName             string `xml:"networkName,attr,omitempty"`
-	TaskHREF                string `xml:"task,attr,omitempty"`
-	TaskStatusName          string `xml:"taskStatusName,attr,omitempty"`
-	TaskDetails             string `xml:"taskDetails,attr,omitempty"`
-	TaskStatus              string `xml:"TaskStatus,attr,omitempty"`
```

Comment:
Difficult to see the diff, as formatting changed, but as I understand these fields got removed in API 29.0:
- VdcEnabled
- VAppParentHREF
- VAppParentName
- HighestSupportedVersion
- VmToolsVersion
- TaskHREF
- TaskStatusName
- TaskDetails
- TaskStatus

Few questions:
1. Do we use them in Terraform?
2. Where did the task go?

CC: @dataclouder @Didainius

Reply:
I run test with terraform - no complains on these changes.

Reply:
Where was not task in version 27.0 -> https://code.vmware.com/apis/72/vcloud-director/doc/doc/types/QueryResultVMRecordType.html

Reply:
VAppParentHREF and VAppParentName change to ContainerName and ContainerId

Reply:
VmToolsVersion -> VmToolsStatus

Reply:
could be that HighestSupportedVersion - > HardwareVersion

Reply:
Again a list about removed fields in changelog may make sense for someone

Reply:
added

Reply:
1.) I suggest making changelog even clearer, and instead of only listing what has been removed and added, list what has been renamed too. I mean:
```
* `QueryResultVMRecordType` fields adjusted to accommodate API v27.0 to v29.0 upgrade as follows:
** Renamed fields: `VmToolsVersion` -> `VmToolsStatus`, ...
** Removed fields: ...
** Added fields: ...
```

This way users (and we) will have know-how **how** this type transformed from older to newer API version.

2.) Do I understand correctly that other than QueryResultVMRecordType, GatewayConfiguration, VMGeneralParams and NetworkConfiguration, other types' definitions haven't changed?

3.) If that's it, then I suggest adding those small pointer changes made in the other three types to changelog as well. The goal would be to have full write-up of the type changes caused by this API upgrade.

Reply:
Updated